### PR TITLE
Concepts draft

### PR DESCRIPTION
**Pull request by michelle-luna**

### CODE REVIEW

File: jekyll/_cci2/concepts.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+*[Basics]({{ site.baseurl }}/2.0/concepts/) > Concepts*
+
+This document provides an overview of the concepts used in CircleCI 2.0 in the following sections:
+
+* TOC
+{:toc}
+
+## Steps
+
+Steps are actions that need to be taken to perform your build. ![step illustration]( {{ site.baseurl }}/assets/img/docs/concepts_step.png) Steps are usually a collection of executable commands. For example, the `checkout` step checks out the source code for a job over SSH. Then, the `run` step executes the `make test` command using a non-login shell by default.
```
Comment (KunalJain) [inline jekyll/_cci2/concepts.md]: Build -> job.

File: jekyll/_cci2/concepts.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+   build3:
+     macos: # Specifies a macOS virtual machine with Xcode version 9.0
+       xcode: "9.0"       
+ ...          
+ ```
+
+
+## Jobs
+
+Jobs are a collection of steps. The Job Space is all of the containers being run by an executor (`docker`, `machine`, or `macos`) for the current job.
```
Comment (KunalJain) [inline jekyll/_cci2/concepts.md]: ` The Job Space is all of the containers being run by an executor (`docker`, `machine`, or `macos`) for the current job.` -> Not sure about this.

File: jekyll/_cci2/concepts.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    steps:
+      - restore_cache: # Restores the cached dependency.
+          key: v1-repo-{{ .Environment.CIRCLE_SHA1 }}       
+```
+
+{% endraw %}
+
+## Workflows
+
+Workflows define a list of jobs and their run order. Within the CI/CD industry, this feature is also referred to as Pipelines. It is possible to run jobs in parallel, sequentially, on a schedule, or with a  manual gate using an approval job.
```
Comment (KunalJain) [inline jekyll/_cci2/concepts.md]: `Within the CI/CD industry, this feature is also referred to as Pipelines.` Maybe we skip this for now.

File: jekyll/_cci2/concepts.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+           - build1 # wait for build1 job to complete successfully before starting
+           # see circleci.com/docs/2.0/workflows/ for more examples.
+      - build3:
+        requires:
+           - build1 # wait for build1 job to complete successfully before starting
+           # run build2 and build3 in parallel to save time.
+```
+{% endraw %}
+
+### Workspaces and Artifacts
```
Comment (KunalJain) [inline jekyll/_cci2/concepts.md]: Do you want to compare `workspaces` to `caches` or `artifacts`?

File: jekyll/_cci2/concepts.md (hunk 1 above)
Comment (michelle-luna) [reply jekyll/_cci2/concepts.md]: thanks, fixed

File: jekyll/_cci2/concepts.md (hunk 2 above)
Comment (michelle-luna) [reply jekyll/_cci2/concepts.md]: I removed it for now and replaced it with "Jobs are a collection of steps and each job must declare an executor that is either `docker`, `machine`, or `macos`. Machine includes a default image if not specified, for Docker and macOS, you must also declare an image."

File: jekyll/_cci2/concepts.md (hunk 3 above)
Comment (michelle-luna) [reply jekyll/_cci2/concepts.md]: thanks, removed

File: jekyll/_cci2/concepts.md (hunk 4 above)
Comment (michelle-luna) [reply jekyll/_cci2/concepts.md]: yes! I've added the comparison table at the end with a link to our blog series on these.

**Pull request merged**